ember-highcharts/src/components/high-charts.ts

```diff
@@ -202,81 +203,66 @@ export default class HighCharts<
    * Dynamically imports the necessary pieces from Highcharts, based on chart type and options.
    */
   async _importHighchartsDeps() {
+    let tmpModule;
+
     if (this.args.mode === 'Map') {
-      Highcharts = await waitForPromise(import('highcharts/highmaps'));
+      tmpModule = await waitForPromise(import('highcharts/modules/map'));
     } else if (this.args.mode === 'StockChart') {
-      Highcharts = await waitForPromise(import('highcharts/highstock'));
+      tmpModule = await waitForPromise(import('highcharts/modules/stock'));
     } else {
-      Highcharts = await waitForPromise(import('highcharts'));
+      tmpModule = await waitForPromise(import('highcharts'));
     }
 
-    const Accessibility = await waitForPromise(
-      import('highcharts/modules/accessibility'),
-    );
-    Accessibility.default(Highcharts);
+    // This one is needed because in
+    // in webpack builds we will have module contents directly in `tmpModule`
+    // in vite builds we will have module contents in `tmpModule.default`
+    // @ts-expect-error No idea what TS wants
+    this.highchartsInstance = tmpModule.default || tmpModule;
+
+    await waitForPromise(import('highcharts/modules/accessibility'));
 
     // 3d support
     if (this.args.chartOptions?.chart?.options3d) {
-      const Boost = await waitForPromise(import('highcharts/modules/boost'));
-      Boost.default(Highcharts);
-      const Highcharts3d = await waitForPromise(
-        import('highcharts/highcharts-3d'),
-      );
-      Highcharts3d.default(Highcharts);
+      await waitForPromise(import('highcharts/modules/boost'));
+      await waitForPromise(import('highcharts/highcharts-3d'));
     }
 
     // Drilldown support
     if (this.args.chartOptions?.drilldown) {
-      const Drilldown = await waitForPromise(
-        import('highcharts/modules/drilldown'),
-      );
-      Drilldown.default(Highcharts);
+      await waitForPromise(import('highcharts/modules/drilldown'));
     }
 
     if (this.args.chartOptions?.chart?.type === 'funnel') {
-      const Funnel = await waitForPromise(import('highcharts/modules/funnel'));
-      Funnel.default(Highcharts);
+      await waitForPromise(import('highcharts/modules/funnel'));
     }
 
     if (this.args.chartOptions?.chart?.type === 'heatmap') {
-      const Heatmap = await waitForPromise(
-        import('highcharts/modules/heatmap'),
-      );
-      const More = await waitForPromise(import('highcharts/highcharts-more'));
-      More.default(Highcharts);
-      Heatmap.default(Highcharts);
+      await waitForPromise(import('highcharts/modules/heatmap'));
```

Comment:
Just importing modules is all we need now? We don't need to tell Highcharts about them like: `Heatmap.default(Highcharts);`?

Reply:
That was why I was talking about the extra boilerplate [as mentioned here](https://www.highcharts.com/docs/getting-started/version-12) if we want to support <v12. I created a function `_preV12MaybeImport` that takes care of this and in my tests both v10 and v12 works.